src/bridge/IrcBridge.ts

```diff
@@ -1072,8 +1072,13 @@ export class IrcBridge {
         if (userIds) {
             for (const userId of userIds) {
                 this.activityTracker.setLastActiveTime(userId);
-                this.dataStore.updateLastSeenTimeForUser(userId).catch((ex) => {
-                    log.warn(`Failed to bump last active time for ${userId} in database`, ex);
+                this.dataStore.getFirstSeenTimeForUser(userId).then((when) => {
```

Comment:
I'd make this it's own function inside datastore personally, and have the logic for setting first seen time handled within. We might even be able to wrap the postgres stuff into one statement.